commands/remove.rb

```diff
@@ -34,17 +45,42 @@ def self.remove(pkg, verbose)
     # Remove the files and directories installed by the package.
     unless pkg.is_fake?
       Dir.chdir CREW_CONFIG_PATH do
-        # Remove all files installed by the package.
+        # Remove all files installed by the package in CREW_PREFIX and
+        # HOME.
+        # Exceptions:
+        # 1. The file exists in another installed package.
+        # 2. The file is in one of the filelists for packages in
+        # CREW_ESSENTIAL_FILES, or a dependendent package of
+        # CREW_ESSENTIAL_PACKAGES.
         flist = File.join(CREW_META_PATH, "#{pkg.name}.filelist")
         if File.file?(flist)
-          File.foreach(flist, chomp: true) do |line|
-            next unless line.start_with?(CREW_PREFIX)
-            if system("grep --exclude #{pkg.name}.filelist -Fxq '#{line}' ./meta/*.filelist")
-              puts "#{line} is in another package. It will not be removed during the removal of #{pkg.name}.".orange
-            else
-              puts "Removing file #{line}".yellow if verbose
-              FileUtils.remove_file line, exception: false
-            end
+          # When searching for files to delete we exclude the files from
+          # all packages and dependent packages of CREW_ESSENTIAL_PACKAGES.
+          essential_deps_exclude_froms = essential_deps.map { |i| File.file?("#{File.join(CREW_META_PATH, i.to_s)}.filelist") ? "--exclude-from=#{File.join(CREW_META_PATH, i.to_s)}.filelist" : '' }.join(' ')
```

Comment:
Also, should we maybe move this logic into a separate private function in `commands/remove.rb`?

Reply:
i.e. have a function which takes the package name and list of essential dependencies and returns the `unique_to_package_files` and `package_files_that_overlap` variables.

Reply:
Honestly maybe it should actually go in package or package_helpers?

I'm having to use it in the testing file for remove I am trying to get working...

Reply:
That's also going to get complicated quickly. This logic is tied tightly to excluding files not to delete, in favor of doing it fast, so maybe moving too much to other places might not make sense.

But feel free to refactor...

Reply:
I'm trying to minimise the amount of stuff in `lib/package.rb` because otherwise we're loading all that code as part of a package object which will probably lead to performance headaches in the future-- package_helpers would be fine but I don't see where we'd need to use this code in any other place apart from here?

Reply:
I'm thinking that at least for generating the list of essential dependencies that should only happen when we do a crew update. Maybe we just add it to device.json then and add it to a variable then? Then we can read it whenever we need to access it?

The other stuff, yeah, better to just leave it in remove...

Reply:
I tried moving more of this stuff out, but it made things slower...